### PR TITLE
Do semantic version comparison in the iModelJsLibraryImportsPlugin for webpack

### PR DESCRIPTION
**Pull request by nick4598**

### CODE REVIEW

File: tools/webpack-core/src/plugins/IModeljsLibraryImportsPlugin.ts [hunk 1]
```diff
@@ -6,12 +6,32 @@
...
+    const dashIndexExtension = extensionVersion.indexOf("-");
+    if (dashIndexExtension >= 0 ) {
+      console.warn("This extension has a dependency on a dev version. Use in prod with caution.");
+      extensionVersion = extensionVersion.substring(0, dashIndexExtension);
+    }
+    if (dashIndexApp >= 0) appVersion = appVersion.substring(0, dashIndexApp);
+    const appVersionSplit = appVersion.split(".").map((value) => parseInt(value));
+    const extensionVersionSplit = extensionVersion.split(".").map((value) => parseInt(value));
+    if (appVersionSplit.length !== 3)
+      throw new Error("An iModel.js Shared Library is not versioned correctly. Expected 3 numbers separated by . for the version");
```
Comment (calebmshafer) [inline tools/webpack-core/src/plugins/IModeljsLibraryImportsPlugin.ts:21]: Can you add a `'`s around the `.`?

File: tools/webpack-core/src/plugins/IModeljsLibraryImportsPlugin.ts [hunk 2]
```diff
@@ -6,12 +6,32 @@
...
 import { ConcatSource } from "webpack-sources";
 import * as utils from "./IModeljsLibraryUtils";
-
 function getLoaderString(pkgName: string, pkgVersion: string) {
-  return `  module.exports = (() => {
+  return `  const semverGte = (appVersion, extensionVersion) => {
+    const dashIndexApp = appVersion.indexOf("-");
+    const dashIndexExtension = extensionVersion.indexOf("-");
+    if (dashIndexExtension >= 0 ) {
+      console.warn("This extension has a dependency on a dev version. Use in prod with caution.");
```
Comment (calebmshafer) [inline tools/webpack-core/src/plugins/IModeljsLibraryImportsPlugin.ts:14]: Can we switch this to be a bit more explicit about the possible complications?

`The Extension '${Extension Name}' depends on a nightly version of iModel.js Shared Libraries.  Use in production with caution as stability is not guaranteed.`

**Review by calebmshafer — CHANGES_REQUESTED** [review]
Couple comments on the messaging but otherwise LGTM.

File: tools/webpack-core/src/plugins/IModeljsLibraryImportsPlugin.ts (hunk 1 above)
Comment (nick4598) [reply tools/webpack-core/src/plugins/IModeljsLibraryImportsPlugin.ts:21]: like '.' or .'s ?

File: tools/webpack-core/src/plugins/IModeljsLibraryImportsPlugin.ts (hunk 2 above)
Comment (nick4598) [reply tools/webpack-core/src/plugins/IModeljsLibraryImportsPlugin.ts:14]: Do you know how I'd get access to the extension name?

Comment (calebmshafer) [reply tools/webpack-core/src/plugins/IModeljsLibraryImportsPlugin.ts:14]: Not sure the best way to get it so lets skip it in this PR.

**Comment by calebmshafer** [comment]
@Mergifyio backport release/2.10.0

File: tools/webpack-core/src/plugins/IModeljsLibraryImportsPlugin.ts (hunk 1 above)
Comment (nick4598) [reply tools/webpack-core/src/plugins/IModeljsLibraryImportsPlugin.ts]: Changed

**Review by calebmshafer — APPROVED** [review]
LGTM

**Pull request merged**